flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/SplitHelpers.java

```diff
@@ -113,7 +113,7 @@ public static List<IcebergSourceSplit> createSplitsFromTransientHadoopTable(
               null,
               properties);
       final GenericAppenderHelper dataAppender =
-          new GenericAppenderHelper(table, FileFormat.PARQUET, temporaryFolder);
+          new GenericAppenderHelper(table, FileFormat.PARQUET, temporaryFolder.getRoot().toPath());
```

Comment:
why is this change needed? shouldn't `temporaryFolder` be already a `Path`?

Reply:
I believe this is caused due to https://github.com/apache/iceberg/pull/10657/files#r1668408135

Reply:
this change was reverted, pls see my response at the next comment